src/librustdoc/clean/types.rs

```diff
@@ -1860,13 +1860,22 @@ crate enum Variant {
 crate struct Span(rustc_span::Span);
 
 impl Span {
+    /// Wraps a [`rustc_span::Span`]. In case this span is the result of a macro expansion, the
+    /// span will be updated to point to the macro invocation instead of the macro definition.
+    ///
+    /// (See rust-lang/rust#39726)
     crate fn from_rustc_span(sp: rustc_span::Span) -> Self {
-        // Get the macro invocation instead of the definition,
-        // in case the span is result of a macro expansion.
-        // (See rust-lang/rust#39726)
         Self(sp.source_callsite())
     }
 
+    /// Unless you know what you're doing, use [`Self::from_rustc_span`] instead!
+    ///
+    /// Contrary to [`Self::from_rustc_span`], this constructor wraps the span as is and don't
+    /// perform any operation on it, even if it's from a macro expansion.
```

Comment:
Maybe this would provide a bit more information?

```suggestion
    /// Unlike [`Self::from_rustc_span`], this function wraps the span as-is and doesn't
    /// clean the span at all. For example, spans resulting from macro expansions will
    /// vary based on whether you use [`Self::from_rustc_span`] or this function.
```

Also, I think a more descriptive function name could help, such as `wrap_raw`. That would indicate to callers that it may not be what they want.

Reply:
Personally I think this is way overkill - the body of both functions is a single line each and you can clearly see the difference by reading the source. `wrap_raw` seems fine though.

Reply:
> the body of both functions is a single line each and you can clearly see the difference by reading the source

It may not be obvious to new contributors that cleaning of the span is necessary, so they may think that wrapping it without any modifications is fine. But I think it could be reasonable to just say

```markdown
This function doesn't clean the span at all.
Compare with [`Self::from_rustc_span`]'s body to see the difference.
```

Reply:
Also, it might be better to rename `from_rustc_span` to `new`, to make it clear that it is the standard constructor.

Reply:
Ok, changing the name and updating the doc then!

Reply:
> It may not be obvious to new contributors that cleaning of the span is necessary, so they may think that wrapping it without any modifications is fine.

I don't think it's reasonable to expect new contributors to do everything right on the very first try. As long as it's clear to the reviewer (i.e. the function name is descriptive) I don't think we need 3 paragraphs of description.